app/controllers/api/v0/analytics_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Api
+  module V0
+    class AnalyticsController < ApplicationController
+      def totals
+        user = get_authenticated_user!
+
+        data = if params[:organization_id]
+                 org = Organization.find_by(id: params[:organization_id])
+                 not_authorized unless org && belongs_to_org?(user, org)
+
+                 AnalyticsService.new(organization: org).totals
+               else
+                 AnalyticsService.new(user: user).totals
+               end
+        render json: data.to_json
+      end
+
+      def historical
+        raise ArgumentError if params[:start].blank?
+
+        user = get_authenticated_user!
+
+        data = if params[:organization_id]
+                 org = Organization.find_by(id: params[:organization_id])
+                 not_authorized unless org && belongs_to_org?(user, org)
+
+                 AnalyticsService.new(organization: org, start: params[:start], end: params[:end]).stats_grouped_by_day
+               else
+                 AnalyticsService.new(user: user, start: params[:start], end: params[:end]).stats_grouped_by_day
+               end
+        render json: data.to_json
+      end
+
+      def past_day
+        user = get_authenticated_user!
+
+        data = if params[:organization_id]
+                 org = Organization.find_by(id: params[:organization_id])
+                 not_authorized unless org && belongs_to_org?(user, org)
+
+                 AnalyticsService.new(organization: org, start: DateTime.current - 1.day).stats_grouped_by_day
+               else
+                 AnalyticsService.new(user: user, start: DateTime.current - 1.day).stats_grouped_by_day
+               end
+        render json: data.to_json
+      end
+
+      private
+
+      def get_authenticated_user!
+        api_token = ApiSecret.find_by(secret: params[:api_token])
```

Comment:
Notice that here you're asking for the API token as a parameter which means that it's probably going to end up in logs or in URLs somewhere. It simplifies developers life because they have to type less with curl but it's a potential leakage of information (being the token personal for each user). The alternative would be to use a custom HTTP header to retrieve the API token

Reply:
Thanks for the tip! I updated it to use a custom HTTP header `API_KEY`.